src/main/java/me/zeroeightsix/kami/util/astaralg/MyAstarAlg.java

```diff
@@ -0,0 +1,152 @@
+package me.zeroeightsix.kami.util.astaralg;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+
+import java.util.*;
+
+public class MyAstarAlg {
+
+    /**
+     * Created by fred41 on 18/07/2020.
+     *
+     * https://en.wikipedia.org/wiki/A*_search_algorithm
+     * Variable names are taken from the page as well
+     */
+
+    private static PriorityQueue<MyAstarNode> openlist;
+    private static HashMap<BlockPos, MyAstarNode> closedlist;
+    private static Vec3i targetVec;
+    private static int addlayer;
+    private static double lowestf;
+    private static MyAstarNode lowestfNode;
+
+    private static MyAstarGrid grid;
+
+    public static List<MyAstarNode> findPath(BlockPos start, BlockPos target, int adiitionallayer) {
+        grid = new MyAstarGrid();
+        MyAstarAlg.addlayer = adiitionallayer;
+        lowestf = Double.POSITIVE_INFINITY;
+
+        closedlist = new HashMap<>();
+        targetVec = new Vec3i(target.getX(), target.getY(), target.getZ());
+        openlist = new PriorityQueue<>(Comparator.comparingDouble(o -> o.f));
+        MyAstarNode startnode = new MyAstarNode(start, adiitionallayer);
+        startnode.f = Math.pow(startnode.pos.getX() - targetVec.getX(), 2) + Math.pow(startnode.pos.getZ() - targetVec.getZ(), 2);
+        lowestfNode = startnode;
+        grid.nodes.put(start, startnode);
+        openlist.add(startnode);
+        int i = 0;
+        do {
+            i++;
+            MyAstarNode currentnode = openlist.remove();
+            if (currentnode.pos.equals(target)) {
+                Astarpathfinder.nopathfound = 0;
+                return buildPath(currentnode);
+            } else {
+                closedlist.put(currentnode.pos, currentnode);
+                expandNode(currentnode);
+            }
+        } while (!openlist.isEmpty() && i < 7000);
+        if (i >= 7000) {
+            System.out.println("no short path");
+        }
+        if (Math.sqrt(lowestfNode.pos.distanceSq(target)) > 10) {
+            Astarpathfinder.nopathfound++;
+        }
+        return buildPath(lowestfNode);
```

Comment:
Formatting is pretty much non existent in these files :upside_down_face:

Reply:
i did auto reformat i iirc but feel free to modifie it to your liking